src/nupic/engine/RegionImpl.cpp

```diff
@@ -230,7 +230,11 @@ void RegionImpl::setParameterArray(const std::string& name, Int64 index,const Ar
       break;
     }
 
-    NTA_ASSERT(rc == 0) << "getParameterArray - failure to get parameter '" << name << "' on node of type " << getType();
+    if (rc != 0)
```

Comment:
I still think `NTA_CHECK` would make more sense than `NTA_THROW` inside an if statement. Are you saying that would deviate from convention in this particular file?